contrib/babelfishpg_common/sql/upgrades/babelfishpg_common--2.5.0--2.6.0.sql

```diff
@@ -3,10 +3,189 @@
 
 SELECT set_config('search_path', 'sys, '||current_setting('search_path'), false);
 
+-- Drops an object if it does not have any dependent objects.
+-- Is a temporary procedure for use by the upgrade script. Will be dropped at the end of the upgrade.
+-- Please have this be one of the first statements executed in this upgrade script. 
+CREATE OR REPLACE PROCEDURE babelfish_drop_deprecated_object(object_type varchar, schema_name varchar, object_name varchar) AS
+$$
+DECLARE
+    error_msg text;
+    query1 text;
+    query2 text;
+BEGIN
+
+    query1 := pg_catalog.format('alter extension babelfishpg_tsql drop %s %s.%s', object_type, schema_name, object_name);
+    query2 := pg_catalog.format('drop %s %s.%s', object_type, schema_name, object_name);
+
+    execute query1;
+    execute query2;
+EXCEPTION
+    when object_not_in_prerequisite_state then --if 'alter extension' statement fails
+        GET STACKED DIAGNOSTICS error_msg = MESSAGE_TEXT;
+        raise warning '%', error_msg;
+    when dependent_objects_still_exist then --if 'drop view' statement fails
+        GET STACKED DIAGNOSTICS error_msg = MESSAGE_TEXT;
+        raise warning '%', error_msg;
+end
+$$
+LANGUAGE plpgsql;
+
 CREATE OR REPLACE FUNCTION sys.varbinarybinary (sys.BBF_VARBINARY, integer, boolean)
 RETURNS sys.BBF_BINARY
 AS 'babelfishpg_common', 'binary'
 LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
 
+-- (sys.BBF_BINARY AS sys.VARCHAR)
+DROP CAST IF EXISTS(sys.BBF_BINARY AS sys.VARCHAR);
```

Comment:
what if there is dependent objects? Add test cases, 

-- prepare
create view test as select cast (some_binval as varchar)

-- upgrade to latest 

-- verify 
select * from test